# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSContext.cpp
# Commit: d8f770d123f2
# Full Hash: d8f770d123f29e774262abeebf29bf062760df41
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-12 04:04:10
# Regressor Bug: 1634135
# File Overlap Count: 2
# Description:
#   Bug 1634135: Don't leak Isolate r=mgaudet
#   
#   ASAN found a leak.
#   
#   We destroy the isolate with a level of indirection through RegexpAPI so that JSContext doesn't have to be able to see the full definition of Isolate.
# ==============================================================================

diff -r 3f680457842f -r d8f770d123f2 js/src/vm/JSContext.cpp
--- a/js/src/vm/JSContext.cpp	Sun May 10 16:00:26 2020 +0000
+++ b/js/src/vm/JSContext.cpp	Sun May 10 16:00:27 2020 +0000
@@ -974,6 +974,10 @@
   }
 #endif
 
+#ifdef ENABLE_NEW_REGEXP
+  irregexp::DestroyIsolate(isolate.ref());
+#endif
+
   js_delete(atomsZoneFreeLists_.ref());
 
   TlsContext.set(nullptr);
